raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```
@@ -243,8 +243,9 @@ private boolean maybeUpdateHighWatermark() {
                         );
                         return true;
                     } else if (highWatermarkUpdateOffset < currentHighWatermarkMetadata.offset) {
-                        log.error("The latest computed high watermark {} is smaller than the current " +
-                                "value {}, which suggests that one of the voters has lost committed data. " +
+                        log.warn("The latest computed high watermark {} is smaller than the current " +
+                                "value {}, which should only happen when voter set membership changes. If the voter " +
+                                "set has not changed this suggests that one of the voters has lost committed data. " +
```

In general we should avoid having `error` or `warn` log messages. In most cases we want to throw an exception if there is an error.

In this case, this state is expected but infrequent so `log.info` seems appropriate.